### PR TITLE
Add dependency guard plugin

### PR DESCRIPTION
**Pull request by jdkoren**
This adds the [Dependency Guard][1] gradle plugin and also commits baseline files for all `*RuntimeClasspath` configurations. Subsequent PRs will incorporate verification against these baselines in CI.

[1]: https://github.com/dropbox/dependency-guard

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -24,6 +24,7 @@ plugins {
     id("jacoco")
     alias(libs.plugins.nowinandroid.android.application.firebase)
     id("com.google.android.gms.oss-licenses-plugin")
+    alias(libs.plugins.dependencyGuard)
```
Comment (JoseAlcerreca) [inline app/build.gradle.kts:27]: Should it be part of the Android Application convention plugin?

Comment (jdkoren) [reply app/build.gradle.kts:27]: I had thought about that briefly. The only other app is the catalog, and that one literally is just UI, so I'm not sure how valuable that would be. What do you think?

File: app/build.gradle.kts [hunk 2]
```diff
@@ -133,3 +134,14 @@ dependencies {
     kspTest(libs.hilt.compiler)
 
 }
+
+dependencyGuard {
+
```
Comment (SimonMarquis) [inline app/build.gradle.kts:139]: Do we need to track all configuration? I would be in favor of only tracking the production build `prodReleaseRuntimeClasspath`.

File: app-nia-catalog/dependencies/releaseRuntimeClasspath.txt
Comment (SimonMarquis) [inline app-nia-catalog/dependencies/releaseRuntimeClasspath.txt:1]: This should probably either be removed, or configured correctly in the corresponding build.gradle.kts file of `:app-nia-catalog`.

File: app/build.gradle.kts (hunk 1 above)
Comment (jdkoren) [reply app/build.gradle.kts]: Added to the Android Application convention plugin

File: app/build.gradle.kts (hunk 2 above)
Comment (jdkoren) [reply app/build.gradle.kts]: Done

File: app-nia-catalog/dependencies/releaseRuntimeClasspath.txt
Comment (jdkoren) [reply app-nia-catalog/dependencies/releaseRuntimeClasspath.txt:1]: Done. Not sure how that happened :sweat_smile:

**Review by SimonMarquis — COMMENTED** [review]
LGTM 👍

**Review by tunjid — APPROVED** [review]
LGTM

**Pull request merged**